src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Models/OmniSharpVSCodeActionContext.cs

```diff
@@ -1,14 +1,17 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Razor.LanguageServer.Serialization;
 using Newtonsoft.Json;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 using OmniSharp.Extensions.LanguageServer.Protocol.Serialization;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models
 {
-    internal class ExtendedCodeActionContext : CodeActionContext
+    internal record OmniSharpVSCodeActionContext : CodeActionContext
     {
+        public static readonly PlatformExtensionConverter<CodeActionContext, OmniSharpVSCodeActionContext> JsonConverter = new();
+
         [Optional]
         [JsonProperty("_vs_selectionRange")]
         public Range SelectionRange { get; set; }
```

Comment:
Shouldn't all of these by changed to be `init;` rather than `set;`? Just the fact that something is a `record` doesn't always make it immutable.

(This comment applies to every property in every record in this PR)

Reply:
> Shouldn't all of these by changed to be `init;`

Love it, yes!